internal/errors/codes.go

```diff
@@ -0,0 +1,8 @@
+package errors
+
+const (
+	ErrDatabaseConnection ErrorCode = "MICROSERVICE_001"
+	ErrDatabaseQuery      ErrorCode = "MICROSERVICE_002"
+	ErrCacheConnection    ErrorCode = "MICROSERVICE_003"
+	// Add more error codes as needed
```

Comment:
Instead of MICROSERVICE_001, it should be microservice specific pattern.
Since this is department service, I think we can follow something like 'DEPT-001', 'DEPT-002', .etc

Reply:
Got it. Renamed the error codes as suggested.